presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -823,7 +824,8 @@ public synchronized void finishInsertIntoExistingPartition(
                                     merge(currentStatistics, statisticsUpdate),
                                     statisticsUpdate),
                             context,
-                            identity));
+                            identity,
+                            Optional.empty()));
```

Comment:
You can pass `session` since you have it.

Also, if we pass `session` we do not need to pass `identity`, since session contains it.
Please change the Action to take session instead of identity.

Please do this in a separate refactor commit, so that this "mechanical"  change is separate from "logic" change adding support for the auto.purge.